### ISSUE TITLE
Please consider moving it to Processing

### ISSUE THREAD

**Issue by pcav** [juliepierson/DissolveWithStats#2]
As a subplugin, so it can be used in more complex workflow

**Comment by bstroebl** [comment]
There is already a [ticket](http://hub.qgis.org/issues/13654) open for improving dissolve in processing

**Comment by juliepierson** [comment]
Ok, nice to have a ticket listing all improvements for the dissolve tool. Maybe my code can be reused somehow, but I'm not sure I could help for this.

**Comment by pcav** [comment]
Actually, moving your code under Processing is quite easy. Could you please ask in qgis-dev mailing list? I'm sure @volaya can help you.

**Comment by juliepierson** [comment]
That's cool ! I'll try and do that next week then, thanks Paolo.

**Comment by juliepierson** [comment]
Hi, I made a [pull request](https://github.com/qgis/QGIS-Processing/pull/70) to solve this issue about a month ago but it's not been accepted yet. Should I do something else ?

**Comment by pcav** [comment]
I see, thanks. QGIS has now moved to core, so please issue your PR against it:
https://github.com/qgis/QGIS/tree/master/python/plugins/processing

**Comment by rayrabcc** [comment]
Esse problema já foi solucionado? como faço para usar esse qgis em um programa que estou escrevendo em python?

**Comment by juliepierson** [comment]
Hi, you can install the "dissolve with stats" plugin. IS it ok for you ?

**Comment by rayrabcc** [comment]
Yes, but I can do use in pyqgis? I have a script and i need use plugin Dissolve with stats. 

'''import DissolveWithStats
import qgis
import qgis.core
import os
import processing
import sys
import qgis.gui
import qgis.analysis
import qgis.utils
import osgeo
import math
#Dissolve with stats
diss = r'C:\GENTE_HOMEOFFICE\Numeracao de Quadras e Lotes\Shapes_Parciais\dissolve_st.shp'
processing.run("qgis:DissolveWithStat",{'INPUT':inter_shp,'OUTPUT':diss,'FIELD' : ['n_quadra']})'''

When i run, the script have error. 

"C:/PROGRA~1/QGIS3~1.0/apps/qgis/./python/plugins\processing\core\Processing.py", line 127, in runAlgorithm
    raise QgsProcessingException(msg)
_core.QgsProcessingException: Erro: Algoritmo qgis:DissolveWithStat não encontrado"

What is the solution?

**Comment by rayrabcc** [comment]
> Olá, você pode instalar o plug-in "dissolver com estatísticas". Está tudo bem para você ?

Hello, can you help me ?

**Comment by juliepierson** [comment]
Sorry, you would have to install the script from the processing branch (https://github.com/UMR-PASSAGES/DissolveWithStats.git) but it's not up to date and does not work with QGIS 3.4. I'll try to fix it but I haven't got much time right now.  Sorry about that ! But from your example it looks like you just need the dissolve algorithm ? You just have one dissolve field with no statistics ?

**Comment by rayrabcc** [comment]
Hello,no proble,I understand. The script the processing in what version the QGIS? I can use the version for solution my problem. I need have one dissolve field with statistics. My script not considers this problem. But i need the use statistics. 
do you need the details what i need? 
thank you for helping me.

**Comment by rayrabcc** [comment]
Hello, i need very much the solution for this problem.

**Comment by pcav** [comment]
Sorry, your question is not very clear to me. The thread is rather long, could you please resume your problem here?

**Comment by rayrabcc** [comment]
I need using plugin Dissolve with stats in my program in python. I writing a codigo and i using tools the qgis, for finish my program i need the using the plugin. 
Example: processing.run("native:dissolve",{'INPUT':lotes_fg,'OUTPUT':p_q,'FIELD' : ['n_quadra']})
this is example i using dissolve in my script , 
But i need using dissolve with stats, how example. 
Did you undersand?

**Comment by pcav** [comment]
Sorry, still unclear to me. AFAICT this function is already available in QGIS, could you please confirm?
Cheers.

**Comment by rayrabcc** [comment]
i need write a script using dissolve with stats, i need put parameters to use stats in tool dissolve. In the example that i put, i don't using stats. 
I can not search solution for this is my problem in tools the QGIS

**Comment by pcav** [comment]
sorry, I really don't understand. can you, @mazano ? @NyakudyaA

**Comment by NyakudyaA** [comment]
I think this is more a user issue than a QGIS plugin issue. Can we have the user ask this on gis.stackexchange or QGIS dev list

**Comment by juliepierson** [comment]
Hello, I think RayraBellico is trying to use Dissolve with stats in its 
processing version, but unfortunately there's a bug in my extension and 
it doesn't work. I'll try to solve this but I don't know when exactly 
I'll be able to do this, hopefully soon !

Le 23.05.20 à 14:51, mazano a écrit :
